src/sentry/static/sentry/app/views/dashboardsV2/widgetQueries.tsx

```diff
@@ -126,108 +126,86 @@ class WidgetQueries extends React.Component<Props, State> {
-          projects,
-          start: start ? getUtcDateString(start) : undefined,
-          end: end ? getUtcDateString(end) : undefined,
-        });
-        const url = `/organizations/${organization.slug}/eventsv2/`;
-        return doDiscoverQuery<TableData>(api, url, {
-          ...eventView.generateQueryStringObject(),
-          per_page: 5,
-        });
+      let url;
+      const params: any = {
+        per_page: 5,
+      };
+      if (widget.displayType === 'table') {
+        url = `/organizations/${organization.slug}/eventsv2/`;
+      } else if (widget.displayType === 'world_map') {
+        url = `/organizations/${organization.slug}/events-geo/`;
+        delete params.per_page;
+      } else {
+        throw Error('Expected widget displayType to be either table or world_map');
+      }
+
+      return doDiscoverQuery<TableData>(api, url, {
+        ...eventView.generateQueryStringObject(),
+        ...params,
       });
+    });
 
-      let completed = 0;
-      promises.forEach(async (promise, i) => {
-        try {
-          const [data] = await promise;
-          // Cast so we can add the title.
-          const tableData = data as TableDataWithTitle;
-          tableData.title = widget.queries[i]?.name ?? '';
+    let completed = 0;
+    promises.forEach(async (promise, i) => {
+      try {
+        const [data] = await promise;
+        // Cast so we can add the title.
+        const tableData = data as TableDataWithTitle;
+        tableData.title = widget.queries[i]?.name ?? '';
 
-          // Overwrite the local var to work around state being stale in tests.
-          tableResults = [...tableResults, tableData];
+        // Overwrite the local var to work around state being stale in tests.
+        tableResults = [...tableResults, tableData];
 
-          completed++;
-          this.setState(prevState => {
-            return {
-              ...prevState,
-              tableResults,
-              loading: completed === promises.length ? false : true,
-            };
-          });
-        } catch (err) {
-          const errorMessage =
-            err?.responseJSON?.detail || t('An unknown error occurred.');
-          this.setState({errorMessage});
-        }
-      });
-    } else if (widget.displayType === 'world_map') {
-      this.setState({tableResults: []});
-      const promises = widget.queries.map(query => {
-        const eventView = EventView.fromSavedQuery({
-          id: undefined,
-          name: query.name,
-          version: 2,
-          fields: [...query.fields],
-          query: query.conditions,
-          projects,
-          start: start ? getUtcDateString(start) : undefined,
-          end: end ? getUtcDateString(end) : undefined,
-        });
-        const url = `/organizations/${organization.slug}/events-geo/`;
-        return doDiscoverQuery<TableData>(api, url, {
-          ...eventView.generateQueryStringObject(),
+        completed++;
+        this.setState(prevState => {
```

Comment:
I haven't tried it, but not sure how the performance of calling `setState` n times as each promise resolves is better or worse than chaining once from `Promise.all`. If loading doesn't show it anyway, there probably isn't a point to have forEach use an async function and we should just use `all` instead for clarity and to improve perf.

Reply:
`setState()` will batch state changes and flush them in the next update cycle. Using `Promise.all()` would be simpler, but would also delay the results from widgets with multiple queries until all the queries are complete. We don't currently expect there to be many widgets with multiple queries as we've not exposed the UI controls to add additional queries yet.

Reply:
Yeah I know `Promise.all()` delays them all, I was saying if `loading` is blocking anyway since it shows loading state for all of them (it checks against promise length) then it's the same. I just checked though and it looks like loading is just used for a mask in the widget cards and it's not exiting early, so my point is moot 😆